spec/mongoid/clients/sessions_spec.rb

```diff
@@ -0,0 +1,297 @@
+require "spec_helper"
+
+describe Mongoid::Clients::Sessions do
+
+  before(:all) do
+    CONFIG[:clients][:other] = CONFIG[:clients][:default].dup
+    CONFIG[:clients][:other][:database] = 'other'
+    Mongoid::Clients.clients.values.each(&:close)
+    Mongoid::Config.send(:clients=, CONFIG[:clients])
+    Mongoid::Clients.with_name(:other).subscribe(Mongo::Monitoring::COMMAND, EventSubscriber.new)
+  end
+
+  after(:all) do
+    Mongoid::Clients.with_name(:other).close
+    Mongoid::Clients.clients.delete(:other)
+  end
+
+  let(:subscriber) do
+    Mongoid::Clients.with_name(:other).instance_variable_get(:@monitoring).subscribers['Command'].find do |s|
+      s.is_a?(EventSubscriber)
+    end
+  end
+
+  let(:insert_events) do
+    subscriber.started_events.select { |event| event.command_name == :insert }
+  end
+
+  let(:update_events) do
+    subscriber.started_events.select { |event| event.command_name == :update }
+  end
+
+  context 'when a session is used on a model class' do
+
+    context 'when sessions are supported', if: sessions_supported? do
+
+      around do |example|
+        Mongoid::Clients.with_name(:other).database.collections.each(&:drop)
+        subscriber.clear_events!
+        Person.with(client: :other) do
+          example.run
+        end
+        Mongoid::Clients.with_name(:other).database.collections.each(&:drop)
+      end
+
+      context 'when the operations in the session block are all on the class' do
+
+        before do
+          Person.with_session do
+            Person.create
+            Person.create
+          end
+        end
+
+        let(:insert_events) do
+          subscriber.started_events.select { |event| event.command['insert'] }
+        end
+
+        it 'uses a single session id for all operations on the class' do
+          expect(Person.count).to be(2)
+          expect(insert_events).not_to be_empty
+          expect(insert_events.collect { |event| event.command['lsid'] }.uniq.size).to eq(1)
```

Comment:
This doesn't actually test that every command sent a session id. I suggest:
```
lsids_sent = insert_events.collect { |event| event.command['lsid'] }
expect(lsids_sent.size).to eq(2)
expect(lsids_sent.uniq.size).to eq(1)
 ```

Same for the other tests. I also suggest adding a test that runs every operation that a Mongoid class can do to make sure we didn't miss adding a session in some place. Please ignore if that's already the case.

Reply:
OK, good point. I'll change this.